tls/s2n_connection.c

```diff
@@ -201,8 +205,11 @@ int s2n_connection_wipe(struct s2n_connection *conn)
 
     /* Clone the stuffers */
     /* ignore gcc 4.7 address warnings because dest is allocated on the stack */
+    /* pragma gcc diagnostic was added in gcc 4.6 */
+#if GCC_VERSION >= 40600
```

Comment:
I think this needs to be tested in an 
    
    #if defined(__GNUC__)

guard also. Otherwise if the compiler doesn't define __GNUC__ et al, and is also intolerant of bad syntax in the if, the condition will end up expanding to:

    #if ( * 10000 + 100 + ) 

and afaict the C standard says that can be considered an error :(